### DISCUSSION TITLE
desktop target integration

### DISCUSSION THREAD

**Discussion by N7ghtm4r3** [swapnil-musale/KDeviceInfo#90] [General]
Hi, can I integrate in a PR the `desktop` target integration?

**Comment by N7ghtm4r3** [comment]
#51

**Reply by swapnil-musale** [reply]
@N7ghtm4r3 That will be really helpful if you do so.

**Comment by swapnil-musale** [comment]
@N7ghtm4r3 Created new issue https://github.com/swapnil-musale/KDeviceInfo/issues/91 for desktop support integration.

If possible do basic integration for all 3 platform as mentioned in issue or any one them is also fine, it's on you.

Thanks for initiating a discussion, hoping to see your PR on the same

**Reply by N7ghtm4r3** [reply]
Yes sure! I am happy to contribute to the library. My idea is integrate this  [library](https://github.com/oshi/oshi) on the desktop target and fill the current empty information on that target. It Is good for you?

**Reply by swapnil-musale** [reply]
Yea, looks good to me

**Reply by N7ghtm4r3** [reply]
Great! I start the PR as soon as possible

**Comment by N7ghtm4r3** [comment]
I have just created the related PR #92!

**Reply by N7ghtm4r3** [reply]
Some methods are not visible, must be fixed do not merge!

**Comment by N7ghtm4r3** [comment]
Sorry for the previous PR with the error, now there is the completed one #93

**Comment by swapnil-musale** [comment]
Thanks @N7ghtm4r3.
Allow me some time to go through it, if it's overall good will merge

**Comment by N7ghtm4r3** [comment]
Yes absolutely! I would like integrate also the `wasm` target if for you it is good, but I am working on it and I need time to do that

**Comment by N7ghtm4r3** [comment]
Hi, I can confirm I can implement also the `wasm` target if for you is good after that `desktop` target has been integrated

**Comment by swapnil-musale** [comment]
Hey @N7ghtm4r3, Please go ahead with wasm implementation https://github.com/swapnil-musale/KDeviceInfo/issues/51
Sorry for delayed response was busy with some other stuff.

And thanks for proactively involving to contribute to this lib.

**Reply by N7ghtm4r3** [reply]
Hi, no problem! I am happy about the merge and the possibility to implement also the `wasm` target. I try to do it as soon as possible!

**Comment by N7ghtm4r3** [comment]
Hi, I want to tell you tomorrow I'll start the `wasm` target integration

**Reply by swapnil-musale** [reply]
No worries, take your time as per your convenience

**Comment by N7ghtm4r3** [comment]
I, yesterday I have created the pull request, but the merge failed. May have I to do something?
